source/application/GUI/Core/MenuBase.h

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2017- Hourglass Resurrection Team
+ * Hourglass Resurrection is licensed under GPL v2.
+ * Refer to the file COPYING.txt in the project root.
+ */
+
+#pragma once
+
+#define WIN32_LEAN_AND_MEAN
+#include <Windows.h>
+
+#include <cstddef>
+#include <memory>
+#include <vector>
+
+struct MenuData
+{
+    std::vector<BYTE> m_menu;
+    std::vector<std::unique_ptr<MenuData>> m_children;
+};
+
+class MenuBase
+{
+protected:
+    MenuData* m_data;

```

Comment:
Perhaps it's possible to store this as a reference as well? I'm not quite sure.

Reply:
Due to how the inheritance is designed, this is not possible. I will not have the information to assign this variable upon immediate construction.

Reply:
Yeah that's why I'm not sure, maybe it's possible to assign it in the body of the constructor, I forgot.